src/event.rs

```diff
@@ -0,0 +1,26 @@
+use nats::asynk::Message as NatsMessage;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct NatsEvent {
+    /// See: https://docs.rs/nats/latest/nats/asynk/struct.Message.html
+    pub nats_subject: String,
+    pub nats_reply: Option<String>,
+    pub nats_data: Vec<u8>,
+}
+
+impl From<NatsEvent> for String {
+    fn from(event: NatsEvent) -> Self {
+        serde_json::to_string(&event).unwrap()
```

Comment:
Need to use fallible `TryFrom` and properly propagate the error instead of `unwrap`

Reply:
It's fallible now